draft/celery-5-high-level-architecture.rst

```diff
@@ -0,0 +1,1216 @@
+    'max_retries': 3,
+    'interval_start': 0,
+    'interval_step': 0.2,
+    'interval_max': 0.2
+  })
+  def foo():
+    second_operation()
+
+  @task(retry_policy={
+    'max_retries': 10,
+    'interval_start': 0,
+    'interval_step': 5,
+    'interval_max': 120
+  })
+  def bar():
+    first_operation()
+    foo.delay()
+
+Or you can wrap each code section in a try..except clause and call
+:py:meth:`celery.app.task.Task.retry`.
+
+.. code-block:: python
+
+  @task(bind=True)
+  def foo(self):
+    try:
+      # first operation
+    except Exception:
+      self.retry(retry_policy={
+        'max_retries': 10,
+        'interval_start': 0,
+        'interval_step': 5,
+        'interval_max': 120
+      })
+
+    try:
+      first_operation()
+    except Exception:
+      self.retry(retry_policy={
+        'max_retries': 10,
+        'interval_start': 0,
+        'interval_step': 5,
+        'interval_max': 120
+      })
+
+    try:
+      second_operation()
+    except Exception:
+      self.retry(retry_policy={
+        'max_retries': 3,
+        'interval_start': 0,
+        'interval_step': 0.2,
+        'interval_max': 0.2
+      })
+
+Those solutions are unnecessarily verbose. Instead, we could use a with clause
+if all we want to do is retry.
+
+.. code-block:: python
+
+  @task
+  def foo():
+    with retry(max_retries=10, interval_start=0, interval_step=5, interval_max=120):
+      first_operation()
+
+    with retry(max_retries=10, interval_start=0, interval_step=5, interval_max=120):
+      second_operation()
+
+By default messages which cannot be re-published will be stored
+in the :ref:`draft/celery-5-high-level-architecture:messages backlog`.
+
+Implementers may provide other fallbacks such as executing the retried task
+in the same worker or abandoning the task entirely.
+
+Health Checks
+~~~~~~~~~~~~~
+
+Health Checks are used in Celery to verify that a worker is able to
+successfully execute a :ref:`task <draft/celery-5-high-level-architecture:Tasks>`
+or a :ref:`service <draft/celery-5-high-level-architecture:Services>`.
+
+The :ref:`draft/celery-5-high-level-architecture:Scheduler` is responsible
+for scheduling the health checks for execution in each worker after
+each time the configured period of time lapses.
+
+Whenever a health check should be executed the
+:ref:`draft/celery-5-high-level-architecture:Scheduler` instructs the
+:ref:`draft/celery-5-high-level-architecture:Publisher` to send the
+`<health check name>_expired` :term:`Event Message` to each worker's
+:ref:`draft/celery-5-high-level-architecture:Inbox Queue`.
+
+Workers which have tasks subscribed to the event will
+execute all the subscribed tasks in order to determine the state of the
+health check.
+
+Health Checks can handle :term:`Document Messages <Document Message>` as input
+from :ref:`draft/celery-5-high-level-architecture:Ingress Only Data Sources`.
+
+This is useful when you want to respond to an alert from a monitoring system
```

Comment:
Not sure I completely understood this sentence. Could you please explain it a bit further?